### DISCUSSION TITLE
Point cloud raycasting

### DISCUSSION THREAD

**Discussion by iulianiuga** [hjoykim/THREE#52] [Q&A] [answered]
Hi, while examining the threejs example from here https://threejs.org/examples/#webgl_interactive_raycasting_points i found that when raycasting to a point cloud must implement a threshold (raycaster.params.Points.threshold = somevalue).
Otherwise when raycasting to point clouds the response can be wrong (for example i got response evn i click at 20 pixels out of the point cloud)

Is there a way to intoduce that parameter to maximise the raycaster picking precision?

**Comment by hjoykim** [comment, marked as answer]
Hello

I've modified Points class  so that we can specify the threshold value when raycasting PointCloud, and committed the related example.

plese refer to this example. 
And let me know if there are any other problems or improvements.

![image](https://github.com/user-attachments/assets/5541a174-9165-4072-8b0f-ced0d46345a8)

**Reply by iulianiuga** [reply]
Thank you for you help.
Just to be aware, it is possible to use distanceToRay attribute doing the same thing - remove all the intersection points returned on which distanceToRay is greater than some threshold. 
In my case i was needed the closest point to the ray and not the closest to the camera, because you can select points from the points clouds through the spaces between the points. 
Again many thanks for your support
